lua_modules/sntp/sntp.lua

```diff
@@ -0,0 +1,290 @@
+local MAX_SERVER_ATTEMPTS = 2
+local SNTP_TIMEOUT        = 5000
+
+-- Constructor
+--   sc and fc are our Success and Failure Callbacks, resp.
+local new = function(serv, sc, fc, now)
+
+  if     type(serv) == "string" then serv = {serv}
+  elseif serv == nil            then serv =
+    {
+      nil,
+      "1.nodemcu.pool.ntp.org",
+      "2.nodemcu.pool.ntp.org",
+    }
+    local ni = net.ifinfo(0)
+    ni = ni and ni.dhcp
+    serv[1] = ni.ntp_server or "0.nodemcu.pool.ntp.org"
+  elseif type(serv) ~= "table"  then error "Bad server table"
+  end
+
+  if type(sc) ~= "function" and type(sc) ~= "lightfunction" then
+    error "Bad success callback type"
+  end
+  if fc ~= nil and type(fc) ~= "function" and type(fc) ~= "lightfunction" then
+    error "Bad failure callback type"
+  end
+  if now ~= nil and type(now) ~= "function" and type(now) ~= "lightfunction" then
+    error "Bad clock type"
+  end
+  now = now or (rtctime and rtctime.get)
+  if now == nil then error "Need clock function" end
+
+  local self = {}
+  self.servers = serv
+
+  local _tmr -- contains the currently running timer, if any
```

Comment:
You clearly are following some convention so prefixing some -- but not all -- local names with "_".  What is this convention?

Reply:
It is not particularly meaningful any more; the code's been through a lot of refactoring.